OpenRA.Game/Effects/SpriteEffect.cs

```diff
@@ -23,14 +23,14 @@ public class SpriteEffect : IEffect
 		readonly bool visibleThroughFog;
 		readonly bool scaleSizeWithZoom;
 
-		public SpriteEffect(WPos pos, World world, string image, string sequence, string palette, bool visibleThroughFog = false, bool scaleSizeWithZoom = false)
+		public SpriteEffect(WPos pos, World world, string image, string sequence, string palette, int facing = 0, bool visibleThroughFog = false, bool scaleSizeWithZoom = false)
```

Comment:
Well, one thing I had in mind was to later ditch inferior special case effects like `Explosion`, `Smoke` and so on and use `SpriteEffect` for those as well, and I wanted to avoid having to explicitly set `false` for the other two optional parameters. But I guess you're right, downstream compatibility is a bit more important.